### PR TITLE
New README.md that integrates intro, credits and references from /doc/

### PR DESCRIPTION
**Pull request by thisisG**
So this is version 0.01 of the new README, now in Markdown.

The README.md file is also used as the main page for the generated documentation that now works
again (I have however not started replacing references in the source code, that is on some TODO list.).

The documentation now generates html and xml instead of LaTeX due to issues with getting the
right dependencies working. The Doxygen file was updated to a more recent version (1.8.11) and
is now the only dependency for generating the documentation (other than make).

Documentation can now be built with

`make libsrtp2doc`

in the root folder, or with

`make`

when in the /doc/ folder.

In order to get the docs up to speed I thought it'd be a decent idea to crowdsource the editing
a bit.

I will start marking things down for "this sounds outdated" and start replacing items that are clearly
out of date or not applicable any longer. Feel free to mark down (heh...) stuff for
checking/removal/editing or, if applicable, push changes you feel are required to this branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+--------------------------------------------------------------------------------
+
+<a name="libsrtp-overview"></a>
+# libSRTP Overview
+
+libSRTP provides functions for protecting RTP and RTCP.  RTP packets
+can be encrypted and authenticated (`using the srtp_protect()`
+function), turning them into SRTP packets. Similarly, SRTP packets
+can be decrypted and have their authentication verified (using the
+srtp_unprotect() function), turning them into RTP packets. Similar
```
Comment (thisisG) [inline README.md]: Add backticks for function

File: README.md [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+  len = get_rtp_packet(rtp_buffer);
+  srtp_protect(session, rtp_buffer, &len);
+  send_srtp_packet(rtp_buffer, len);
+}
+~~~
+
+--------------------------------------------------------------------------------
+
+<a name="isma-encryption-support"></a>
+# ISMA Encryption Support
```
Comment (thisisG) [inline README.md]: This whole section I think can be removed? Re PR #238 and Issue #247, might need some investigation first

File: README.md [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+
+// set policy to describe a policy for an SRTP stream
+crypto_policy_set_rtp_default(&policy.rtp);
+crypto_policy_set_rtcp_default(&policy.rtcp);
+policy.ssrc = ssrc;
+policy.key  = key;
+policy.next = NULL;
+
+// set key to random value
+crypto_get_random(key, 30);
```
Comment (thisisG) [inline README.md]: Need new alternative here I think - this is removed

File: README.md [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+\-\-gdoi                  | use GDOI key management (disabled at present)
+
+By default, dynamic debugging is enabled and stdout is used for
+debugging. You can use the configure options to have the debugging
+output sent to syslog or the system console. Alternatively, you can
+define `ERR_REPORTING_FILE` in `include/conf.h` to be any other
+file that can be opened by libSRTP, and debug messages will be sent to
+it.
+
+This package has been tested on the following platforms: Mac OS X
```
Comment (thisisG) [inline README.md]: Probably remove this? Not sure if it makes sense to mention specific systems when we got Travis?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+  * The replay window is 128 bits in length, and is hard-coded to this
+    value for now.
+
+--------------------------------------------------------------------------------
+
+<a name="installing-and-building-libsrtp"></a>
+# Installing and Building libSRTP
+
+To install libSRTP, download the latest release of the distribution
+from [github](https://github.com/cisco/libsrtp/releases). You probably
```
Comment (thisisG) [inline README.md]: Maybe specify that it is releases the link points to

File: README.md [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+section in the generated documentation.
+
+Some options that are not included in the specification are supported.
+Most notably, the TMMH authentication function is included, though it
+was removed from the SRTP Internet Draft during the summer of 2002.
+
+Some options that are described in the SRTP specification are not
+supported. This includes
+
+- the Master Key Index (MKI),
```
Comment (thisisG) [inline README.md]: I think we got this very soon, waiting for PR #224

File: README.md [hunk 7]
```diff
@@ -0,0 +1,488 @@
...
+course, other SRTP participants will need to use the key for
+decryption). libSRTP supports this enforcement by detecting the case
+in which a key is used for both inbound and outbound data.
+
+--------------------------------------------------------------------------------
+
+<a name="supported-features"></a>
+## Supported Features
+
+This library supports all of the mandatory-to-implement features of
```
Comment (thisisG) [inline README.md]: Think this needs to be fact checked

Comment (thisisG) [inline README.md]: Mainly due to it stating "defined by the most recent Internet Draft" :-)

File: README.md [hunk 8]
```diff
@@ -0,0 +1,488 @@
...
+SRTP (as defined by the most recent Internet Draft). Some of these
+features can be selected (or de-selected) at run time by setting an
+appropriate policy; this is done using the structure `srtp_policy_t`.
+Some other behaviors of the protocol can be adapted by defining an
+approriate event handler for the exceptional events; see the SRTPevents
```
Comment (thisisG) [inline README.md]: Check that the mentioned section exists in generated docs

File: doc/Makefile.in [hunk 9]
```diff
@@ -18,27 +16,15 @@ VPATH = @srcdir@
 
 version = $(shell cat $(top_srcdir)/VERSION)
```
Comment (thisisG) [inline doc/Makefile.in]: Remember to remove this

File: README.md (hunk 2 above)
Comment (pabuhler) [inline README.md]: kill it

File: README.md (hunk 3 above)
Comment (pabuhler) [inline README.md]: yes see #161, either have a hard coded key or else use openssl crypto random function.

File: README.md (hunk 4 above)
Comment (pabuhler) [inline README.md]: I think we should have a list of "official" tested system, ie a minimal set of system that will be tested before a release. We can add this later as we get to a 2.1 release and see what systems we can get hold of

File: README.md (hunk 5 above)
Comment (pabuhler) [inline README.md]: Good idea

File: README.md (hunk 6 above)
Comment (pabuhler) [inline README.md]: yip so, just ignore for now

File: doc/Makefile.in (hunk 9 above)
Comment (pabuhler) [inline doc/Makefile.in]: where will the version come from?

**Comment by pabuhler** [comment]
Looks good, tested viewing generated html docs at http://htmlpreview.github.io/?https://github.com/pabuhler/libsrtp/blob/t-doc-test/doc/html/index.html , given that it works ok I would remove the xml generation for now.

I suggest we get this checked in so we can review the generated documentation for correctness and fix it under ways.

Some follow up tasks would be to remove crypto_kernal.txt as all that information should be in header files and add the documentation build step to travis to prevent it breaking.

File: doc/Makefile.in (hunk 9 above)
Comment (thisisG) [inline doc/Makefile.in]: Hmm good point. I guess I could re-introduce it for when the docs are generated. Right now it is in "manual mode" in the Doxyfile. Will fix!

**Comment by thisisG** [comment]
I'll remove the xml generation and do the fixups possible with regards to comments tomorrow!

**Pull request merged**